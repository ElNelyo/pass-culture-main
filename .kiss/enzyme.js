// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
