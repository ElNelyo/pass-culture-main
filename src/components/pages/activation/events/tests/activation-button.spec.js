// jest --env=jsdom ./src/components/pages/activation/events/tests/activation-button --watch
import React from 'react'
import { shallow } from 'enzyme'
import ActivationOnlineButton from '../activation-button'

describe('src | components | pages | activation | events | ActivationOnlineButton', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<ActivationOnlineButton />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
