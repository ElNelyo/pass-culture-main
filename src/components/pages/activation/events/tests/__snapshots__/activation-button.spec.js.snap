// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | activation | events | ActivationOnlineButton snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivationOnlineButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="mb36"
        >
          <a
            className="is-block text-center button is-rounded py7 no-background"
            href="https://www.demarches-simplifiees.fr/commencer/activer-mon-compte-pass-culture"
            id="activation-online-button"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="fs18 is-white-text is-bold"
            >
              Activer votre pass en ligne
            </span>
          </a>
        </div>,
        <div
          className="text-center"
        >
          <span
            className="is-italic is-medium fs20"
          >
            -- ou --
          </span>
        </div>,
      ],
      "className": "mb36",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="is-block text-center button is-rounded py7 no-background"
            href="https://www.demarches-simplifiees.fr/commencer/activer-mon-compte-pass-culture"
            id="activation-online-button"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="fs18 is-white-text is-bold"
            >
              Activer votre pass en ligne
            </span>
          </a>,
          "className": "mb36",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="fs18 is-white-text is-bold"
            >
              Activer votre pass en ligne
            </span>,
            "className": "is-block text-center button is-rounded py7 no-background",
            "href": "https://www.demarches-simplifiees.fr/commencer/activer-mon-compte-pass-culture",
            "id": "activation-online-button",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Activer votre pass en ligne",
              "className": "fs18 is-white-text is-bold",
            },
            "ref": null,
            "rendered": "Activer votre pass en ligne",
            "type": "span",
          },
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="is-italic is-medium fs20"
          >
            -- ou --
          </span>,
          "className": "text-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-- ou --",
            "className": "is-italic is-medium fs20",
          },
          "ref": null,
          "rendered": "-- ou --",
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="mb36"
          >
            <a
              className="is-block text-center button is-rounded py7 no-background"
              href="https://www.demarches-simplifiees.fr/commencer/activer-mon-compte-pass-culture"
              id="activation-online-button"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="fs18 is-white-text is-bold"
              >
                Activer votre pass en ligne
              </span>
            </a>
          </div>,
          <div
            className="text-center"
          >
            <span
              className="is-italic is-medium fs20"
            >
              -- ou --
            </span>
          </div>,
        ],
        "className": "mb36",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="is-block text-center button is-rounded py7 no-background"
              href="https://www.demarches-simplifiees.fr/commencer/activer-mon-compte-pass-culture"
              id="activation-online-button"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="fs18 is-white-text is-bold"
              >
                Activer votre pass en ligne
              </span>
            </a>,
            "className": "mb36",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="fs18 is-white-text is-bold"
              >
                Activer votre pass en ligne
              </span>,
              "className": "is-block text-center button is-rounded py7 no-background",
              "href": "https://www.demarches-simplifiees.fr/commencer/activer-mon-compte-pass-culture",
              "id": "activation-online-button",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Activer votre pass en ligne",
                "className": "fs18 is-white-text is-bold",
              },
              "ref": null,
              "rendered": "Activer votre pass en ligne",
              "type": "span",
            },
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="is-italic is-medium fs20"
            >
              -- ou --
            </span>,
            "className": "text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "-- ou --",
              "className": "is-italic is-medium fs20",
            },
            "ref": null,
            "rendered": "-- ou --",
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
